cmd/registry/core/zip.go

```diff
@@ -101,23 +102,24 @@ func UnzipArchiveToMap(b []byte) (map[string][]byte, error) {
 // ZipArchiveOfPath reads the contents of a path into a zip archive.
 // The specified prefix is stripped from file names in the archive.
 // Based on an example published at https://golangcode.com/create-zip-files-in-go/
-func ZipArchiveOfPath(path, prefix string) (buf bytes.Buffer, err error) {
+func ZipArchiveOfPath(path, prefix string, recursive bool) (buf bytes.Buffer, err error) {
 	zipWriter := zip.NewWriter(&buf)
 	defer zipWriter.Close()
 
-	_ = filepath.Walk(path, func(p string, info os.FileInfo, err error) error {
+	err = filepath.WalkDir(path, func(p string, entry fs.DirEntry, err error) error {
```

Comment:
True. I don't love this optimization because the resulting code is
```
return buf, filepath.WalkDir...
```
and this expects readers to know that buf is returned-by-reference and thus the returned value is modified by the expression that follows it. I think people *should* know that but it feels like a Go interview question. Happy to change it if you feel strongly, though.